pkg/memtier/tracker_idlepage.go

```diff
@@ -236,6 +241,8 @@ func (t *TrackerIdlePage) sampler() {
 			t.toSampler = nil
 			return
 		case <-ticker.C:
+			t.regionsMutex.Lock()
+			defer t.regionsMutex.Unlock()
```

Comment:
All deferred function calls are executed when exiting the function. This is probably not what you wish here...